homeassistant/components/fritzbox/climate.py

```diff
@@ -168,6 +174,11 @@ def hvac_mode(self) -> HVACMode:
 
     async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
         """Set new operation mode."""
+        if self.data.lock:
```

Comment:
lock is automatically enabled, when summer or holiday mode is active - so to still be able to distinguish between them, we need to move this after `if self.data.holiday_active or self.data.summer_active:`